api/v1alpha1/argocd_conversion.go

```diff
@@ -28,15 +28,18 @@ func (src *ArgoCD) ConvertTo(dstRaw conversion.Hub) error {
 	// deprecated keycloak configs set in alpha (.spec.sso.image, .spec.sso.version, .spec.sso.verifyTLS, .spec.sso.resources),
 	// override .spec.sso.keycloak in beta
 	if src.Spec.SSO != nil && !reflect.DeepEqual(src.Spec.SSO, &ArgoCDSSOSpec{}) {
-		if src.Spec.SSO.Image != "" || src.Spec.SSO.Version != "" || src.Spec.SSO.VerifyTLS != nil || src.Spec.SSO.Resources != nil {
+		if src.Spec.SSO.Image != "" || src.Spec.SSO.Version != "" || src.Spec.SSO.VerifyTLS != nil || src.Spec.SSO.Resources != nil || (src.Spec.SSO.Keycloak != nil && src.Spec.SSO.Keycloak.Host != "") {
 			if sso.Keycloak == nil {
 				sso.Keycloak = &v1beta1.ArgoCDKeycloakSpec{}
 			}
 			sso.Keycloak.Image = src.Spec.SSO.Image
 			sso.Keycloak.Version = src.Spec.SSO.Version
 			sso.Keycloak.VerifyTLS = src.Spec.SSO.VerifyTLS
 			sso.Keycloak.Resources = src.Spec.SSO.Resources
-			sso.Keycloak.Host = src.Spec.SSO.Keycloak.Host
```

Comment:
We can get rid of this line entirely. The `ArgoCDKeycloakSpec` struct for v1alpha1 & v1beta1 is same so there is no need to perform a explicit conversion. The `ConvertAlphaToBetaSSO` func will automatically  do a implicit conversion without any data loss. https://github.com/argoproj-labs/argocd-operator/blob/a051fbd5c4b04040e90a6356c4826ebea7c6e803/api/v1alpha1/argocd_conversion.go#L285-L295 
I ran some local tests without this line. The conversion of  `.host` field is happening correctly without any issues. We just remove this line and it should be enough to fix the nil panic.

Reply:
Done, thanks @svghadi for reviewing and testing!